http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoMetricsMonitor.scala

```diff
@@ -0,0 +1,342 @@
+      c <- allClusters
+      n <- c.nodepools
+      a <- n.apps
+    } yield Map(
+      AppStatusMetric(c.cloudContext.cloudProvider,
+                      a.appType,
+                      a.status,
+                      getRuntimeUI(a.labels),
+                      getAzureCloudContext(c.cloudContext)
+      ) -> 1
+    )
+
+    allApps.combineAll
+  }
+
+  /** Transforms DB runtimes to RuntimeStatusMetric and computes counts. */
+  private[monitor] def countRuntimesByDbStatus(allRuntimes: List[RuntimeMetrics]): Map[RuntimeStatusMetric, Int] = {
+    val allContainers = for {
+      r <- allRuntimes
+      c <- r.containers if c != RuntimeContainerServiceType.WelderService
+    } yield Map(
+      RuntimeStatusMetric(r.cloudContext.cloudProvider,
+                          c.imageType,
+                          r.status,
+                          getRuntimeUI(r.labels),
+                          getAzureCloudContext(r.cloudContext)
+      ) -> 1
+    )
+
+    allContainers.combineAll
+  }
+
+  /** Performs health checks for Running apps, and transforms to AppHealthMetric. */
+  private[monitor] def countAppsByHealth(
+    allClusters: List[KubernetesCluster]
+  )(implicit ev: Ask[F, AppContext]): F[Map[AppHealthMetric, Int]] = {
+    val allServices = for {
+      c <- allClusters if c.asyncFields.isDefined
+      n <- c.nodepools
+      // Only care about Running apps for health check metrics
+      a <- n.apps if a.status == AppStatus.Running
+      s <- a.appResources.services
+    } yield (c.cloudContext, c.asyncFields.get.loadBalancerIp, a, s.config.name)
+
+    allServices
+      .parTraverseN(parallelism) { case (cloudContext, baseUri, app, serviceName) =>
+        for {
+          ctx <- ev.ask
+          // For GCP just test if the app is available through the Leo proxy.
+          // For Azure impersonate the user and call the app's status endpoint via Azure Relay.
+          isUp <- cloudContext match {
+            case CloudContext.Gcp(project) => appDAO.isProxyAvailable(project, app.appName, serviceName)
+            case CloudContext.Azure(_) =>
+              for {
+                tokenOpt <- samDAO.getCachedArbitraryPetAccessToken(app.auditInfo.creator)
+                token <- F.fromOption(
+                  tokenOpt,
+                  AppCreationException(s"Pet not found for user ${app.auditInfo.creator}", Some(ctx.traceId))
+                )
+                authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, token))
+                relayPath = Uri.unsafeFromString(baseUri.asString) / app.appName.value
+                isUp <- serviceName match {
+                  case ServiceName("wds")      => wdsDAO.getStatus(relayPath, authHeader).handleError(_ => false)
+                  case ServiceName("cbas")     => cbasDAO.getStatus(relayPath, authHeader).handleError(_ => false)
+                  case ServiceName("cbas-ui")  => cbasUiDAO.getStatus(relayPath, authHeader).handleError(_ => false)
```

Comment:
Noob question again, but why do we only separate the ui portion for cbas?

Reply:
Because only cbas-app has a separate UI. The others are just Java services.